spark/src/main/scala/org/apache/spark/sql/comet/CometRowToColumnarExec.scala

```diff
@@ -60,8 +62,17 @@ case class CometRowToColumnarExec(child: SparkPlan)
     val timeZoneId = conf.sessionLocalTimeZone
     val schema = child.schema
 
-    child
-      .execute()
+    val rdd: RDD[InternalRow] = if (child.supportsColumnar) {
+      child
+        .executeColumnar()
+        .mapPartitionsInternal { iter =>
+          iter.flatMap(_.rowIterator().asScala)
+        }
+    } else {
+      child.execute()
+    }
+
+    rdd
```

Comment:
There might be a more efficient way than using a row iterator to write to the row based arrow writer, but as this is mostly for testing/fallback purposes, I didn't try to figure out a faster Spark vector to arrow vector approach. Hopefully someone is able to add complex type support to the Comet parquet reader. If not it could be worth thinking about using the Spark reader as a real fallback use case instead of just for testing purposes.

Reply:
Also not sure if this means the class should be renamed from `RowToColumnar` since this is technically just doing `ColumnarToColumnar`

Reply:
This class is now capable of converting Spark rows or Spark columns to Comet/Arrow columnar format. Perhaps we should consider renaming it. I'm not sure what a good name would be. Some ideas: 

- CometSparkToColumnarExec
- CometConvertFromSparkExec

Reply:
Yeah I agree a new name would probably make sense. Was worried about bloating this PR too much though 😅

Reply:
> it could be worth thinking about using the Spark reader as a real fallback use case instead of just for testing purposes
It might be worthwhile, though I think we would like to prioritize the complex type support in the reader.

Reply:
We can change the name in a future PR

Reply:
I'll write up an issue